lib/global.h

```diff
@@ -9,7 +9,13 @@
 
 #include <glib.h>
 
-#if defined(HAVE_FUNC_ATTRIBUTE_WEAK) && defined(HAVE_TESTS)
+#if defined(__has_attribute)
+#define MC_HAS_ATTRIBUTE __has_attribute
```

Comment:
for legibility, i'd prefer this to be a function-like macro. the branches would be consistent then, too.
if that doesn't work for some reason, a comment should say so.

Reply:
I don't know what you mean. Can you just suggest a fix? As I said, my relationship with macros and C in general is a difficult one to remain politically correct.

Reply:
yep, looking good now.